mart/attack/adversary.py

```diff
@@ -109,6 +110,8 @@ def __init__(
         self.perturber = perturber
         self.composer = composer
         self.optimizer_fn = optimizer
+        if not isinstance(self.optimizer_fn, OptimizerFactory):
```

Comment:
Shall we add this to `mart/models/modular.py` so that we can simplify the config in `mart/configs/optimizer`? That will make it easier to add other off-the-shelf optimizers from PyTorch.

Reply:
Perhaps. But I think that should live in another PR.

Reply:
I will make a PR.